kernel/kernel-impl/src/main/java/org/sakaiproject/ignite/IgniteConfigurationAdapter.java

```diff
@@ -242,33 +277,53 @@ private int dynamicPort(int number) {
         return number;
     }
 
+    /**
+     * When the node is named, the "Ignite home directory" is one level under the
+     * configured IGNITE_HOME / ignite.home path. For example, this name will almost
+     * always be the FDQN of the server and match the serverId property. The usual
+     * path will be of the form:   ${sakai.home}/ignite/[machine-name]
+     */
     private void configureHome() {
-        String path = home;
-        if (StringUtils.isBlank(path)) {
-            // if the root path is not specified use sakai.home/ignite/
-            path = serverConfigurationService.getSakaiHomePath();
-            path = path + File.separator + "ignite";
-        }
-
-        if (StringUtils.isNotBlank(node)) {
-            if (!StringUtils.endsWith(path, File.separator)) {
-                path = path + File.separator;
-            }
-            path = path + node;
+        var base = Path.of(home);
+        var igniteHome = StringUtils.isBlank(node) ? base : base.resolve(node);
+
+        try {
+            Files.createDirectories(igniteHome);
+            if (!Files.isWritable(igniteHome)) throw cannotWriteToIgniteDirectory(igniteHome);
+            home = igniteHome.toAbsolutePath().toString();
+        } catch (IOException e) {
+            throw cannotCreateIgniteDirectory(igniteHome);
         }
+    }
 
-        File igniteHome = new File(path);
-        if (!igniteHome.exists()) igniteHome.mkdirs();
-
-        // return the absolute path
-        home = igniteHome.getAbsolutePath();
+    /**
+     * The default Ignite base directory is ${sakai.home}/ignite/
+     */
+    protected String fallbackBaseDirectory() {
+        return sakai_home.getPathPlus("ignite")
```

Comment:
Note: we use the SysProp here because of the super special status of `sakai.home` and the tricky dependency graph around SakaiProperties. This is definitely better than `System.getProperty`, though.